service/qos/service_level_controller.cc

```diff
@@ -220,6 +220,72 @@ future<> service_level_controller::update_service_levels_cache() {
     });
 }
 
+static std::unordered_set<sstring> get_all_roles(const std::unordered_multimap<sstring, sstring>& hierarchy, const sstring& base_role) {
```

Comment:
`get_all_roles` doesn't sound like a good name for this function. It does not return all roles that exist. Maybe something more descriptive?